sktime/forecasting/base/adapters/_generalised_statsforecast.py

```diff
@@ -26,7 +29,16 @@ class _GeneralisedStatsForecastAdapter(BaseForecaster):
     def __init__(self):
         super().__init__()
 
-        self._forecaster = None
+        self._forecaster = self._instantiate_model()
```

Comment:
I'll prefer if you avoid this. We are initialising the model during fit for all concrete estimators now, and I'd prefer to keep it that way. Here's my suggestion:

1. create a new property storing the model class in all concrete estimators as a string, e.g. "AutoARIMA", "GARCH", etc.
2. in both of the new methods, check if "level" is in `predict` or `predict_in_sample` of `getattr(statsforeast.models, self.model_class)` or not, without initiating an estimator of that class

This is the line failing in tests for MSTL class. I believe not instantiating the estimator will solve that as well, can you confirm @luca-miniati?

Reply:
hm, this seems a bit clunky to me. May I suggest a variation on the idea which avoids a new property?

What I would do is, instead of a string which we have to resolve later, to deal with it similar in the other adapter, `_TslearnAdapter` - using a `_get_statsforecast_class` method, which returns the class, the output can then be used -possibly in another method `_supports_proba` - to inspect the method signatures.

From the extender contract perspective, changing a class import in `get_statsforecast_class` is just as much work as setting a string attribute, and it seems more flexible and transparent.

Reply:
Makes sense, and I suppose initiate forecaster can use this method as well to intiate the instance. It'd be consistent with `tslearn` adapter then.

(I remember a open `statsmodels` issue with `random_state`. Don't remember it very well, and can't find now, but will this sort of handling help it as well?)

Reply:
> Don't remember it very well, and can't find now, but will this sort of handling help it as well?

Here, there wasn't an issue, as it is subsumed under generic deprecation message handling:

https://github.com/sktime/sktime/pull/4623

I don't think the problem is solved, since in `statsmodels` the `random_state` arg can be passed to `kwargs`, which cannot be determined by a signature inspection.

Reply:
> This is the line failing in tests for MSTL class. I believe not instantiating the estimator will solve that as well, can you confirm @luca-miniati?

@yarnabrina I agree, init params in `StatsForecastMSTL` are set after the call to `_GeneralisedStatsForecastAdapter.__init__`. Should work if `_instantiate_model` is moved back to `fit`